keras_cv/layers/preprocessing/channel_shuffle.py

```diff
@@ -0,0 +1,89 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf 
+from tensorflow.keras import layers, backend
+
+class ChannelShuffle(layers.Layer):
+    """ChannelShuffle class for shuffling the channel of RGB image. The expected images 
+    should be [0-255] pixel ranges.
+
+    Input shape:
+        3D (unbatched) or 4D (batched) tensor with shape:
+        `(..., height, width, channels)`, in `"channels_last"` format
+    Output shape:
+        3D (unbatched) or 4D (batched) tensor with shape:
+        `(..., height, width, channels)`, in `"channels_last"` format
+
+    Args:
+        seed:
+            Integer. Used to create a random seed.
+
+    Usage:
+    ```python
+    (images, labels), _ = tf.keras.datasets.cifar10.load_data()
+    channelshuffle = keras_cv.layers.preprocessing.ChannelShuffle()
+    augmented_images = channelshuffle(images)
+    ```
+    """
+    def __init__(
+        self,
+        seed=None,
+        **kwargs
+    ):
+        super().__init__(**kwargs)
+        self.seed = seed
+
+    @tf.function
+    def _channel_shuffling(self, image):
+        x = tf.transpose(image)
+        x = tf.random.shuffle(x, seed=self.seed)
+        return tf.transpose(x)
+
+    def call(self, images, training=None):
+        """call method for the ChannelShuffle layer.
+
+        Args:
+            images: Tensor representing images of shape
+                [batch_size, width, height, channels], with dtype tf.float32 / tf.uint8, or,
+                [width, height, channels], with dtype tf.float32 / tf.uint8
+        Returns:
+            images: augmented images, same shape as input.
+        """
+        if training is None:
```

Comment:
We can just assume `training` is a Python bool.

Reply:
Followed this style https://github.com/keras-team/keras/blob/v2.8.0/keras/layers/preprocessing/image_preprocessing.py#L253-L255

Reply:
Can we standardize the call signature? sometimes we have images in other case we have inputs

Reply:
Until now, in most cases `images` were chosen in [keras-cv](https://github.com/keras-team/keras-cv/tree/master/keras_cv/layers/preprocessing), which makes it reasonable to pick for the CV layers. But I think layers in `keras` have the `inputs` signature.

Reply:
> Followed this style https://github.com/keras-team/keras/blob/v2.8.0/keras/layers/preprocessing/image_preprocessing.py#L253-L255

Going forwards we can assume `training` is a Python bool.

Reply:
@LukeWood 
Can you elaborate on your quote? Did you mean `if training:` instead of  `if training is None:`?

Reply:
Yep!  we don't need the if training is None block anymore.

you can just use training as if it is a python bool.  it will never be None.